### PR TITLE
Fb1436

### PR DESCRIPTION
**Pull request by spireken**
I updated the Book View, Charting, Time and Sales, and Toolbar windows on Dropbox.

Edit: Correction, the Security Input window wasn't changed.

### CODE REVIEW

File: Applications/spire/source/ui/window.cpp [hunk 1]
```diff
@@ -89,260 +58,154 @@ void Window::set_svg_icon(const QString& icon_path,
...
-      (m_body->sizePolicy().horizontalPolicy() != QSizePolicy::Fixed ||
-      m_body->sizePolicy().verticalPolicy() != QSizePolicy::Fixed)) {
-    auto resize_width = scale_width(PADDING_SIZE);
-    if(pos.x() <= resize_width &&
-        m_body->sizePolicy().horizontalPolicy() != QSizePolicy::Fixed) {
-      m_current_active_rect |= ActiveResizeRect::LEFT;
-    } else if(pos.x() >= width() - resize_width &&
-        m_body->sizePolicy().horizontalPolicy() != QSizePolicy::Fixed) {
-      m_current_active_rect |= ActiveResizeRect::RIGHT;
+bool Window::nativeEvent(const QByteArray &eventType, void *message,
```
Comment (kranar) [inline Applications/spire/source/ui/window.cpp]: Fix up typenames.

File: Applications/spire/source/ui/window.cpp [hunk 2]
```diff
@@ -89,260 +58,154 @@ void Window::set_svg_icon(const QString& icon_path,
...
+  if(msg->message == WM_NCCALCSIZE) {
+    *result = 0;
+    return true;
+  } else if(msg->message == WM_NCHITTEST) {
+    auto window_rect = RECT{};
+    GetWindowRect(reinterpret_cast<HWND>(effectiveWinId()), &window_rect);
+    auto x = GET_X_LPARAM(msg->lParam);
+    auto y = GET_Y_LPARAM(msg->lParam);
+    if(m_is_resizeable) {
+      if (x >= window_rect.left &&
```
Comment (kranar) [inline Applications/spire/source/ui/window.cpp]: Remove space after ```if```.

File: Applications/spire/source/ui/window.cpp [hunk 3]
```diff
@@ -89,260 +58,154 @@ void Window::set_svg_icon(const QString& icon_path,
...
+      }
     }
-    auto resize_height = scale_height(PADDING_SIZE);
-    if(pos.y() <= resize_height &&
-        m_body->sizePolicy().verticalPolicy() != QSizePolicy::Fixed) {
-      m_current_active_rect |= ActiveResizeRect::TOP;
-    } else if(pos.y() >= height() - resize_height &&
-        m_body->sizePolicy().verticalPolicy() != QSizePolicy::Fixed) {
-      m_current_active_rect |= ActiveResizeRect::BOTTOM;
+    QPoint pos = m_title_bar->mapFromGlobal({x, y});
```
Comment (kranar) [inline Applications/spire/source/ui/window.cpp]: Always use auto.

**Comment by spireken** [comment]
The builds are Release mode builds.

File: Applications/spire/source/ui/title_bar.cpp [hunk 4]
```diff
@@ -20,17 +20,15 @@ namespace {
...
+TitleBar::TitleBar(const QImage& icon, QWidget* parent)
+    : TitleBar(icon, icon, parent) {}
 
 TitleBar::TitleBar(const QImage& icon, const QImage& unfocused_icon,
-    QWidget* body, QWidget* parent)
-    : QWidget(parent),
-      m_is_dragging(false),
-      m_body(body) {
+    QWidget* parent)
+    : QWidget(parent) {
```
[the remark this answers is not in the document]
Comment (spireken) [reply Applications/spire/source/ui/title_bar.cpp]: I thought this is how we usually indent this. Like in the [Time and Sales window](https://github.com/eidolonsystems/nexus/blob/spire2018/Applications/spire/source/time_and_sales/time_and_sales_window.cpp).

File: Applications/spire/include/spire/ui/window.hpp [hunk 5]
```diff
@@ -39,44 +41,39 @@ namespace Spire {
...
+      void setFixedSize(int width, int height);
+
+      void setFixedSize(const QSize& size);
+
+      void setWindowFlag(Qt::WindowType flag, bool on = true);
+
+      void setWindowFlags(Qt::WindowFlags type);
+
+      //! Connects a slot to the window closed signal.
+      boost::signals2::connection connect_closed_signal(
```
[the remark this answers is not in the document]
Comment (spireken) [reply Applications/spire/include/spire/ui/window.hpp]: We use this to close the testers when the main window is closed.

**Comment by spireken** [comment]
So the QEvent::WinIdEvent event will trigger when the window flags are modified. For whatever reason the window handle doesn't actually change, but the event still triggers.

**Pull request merged**